tools/docker/tests/test_flows.sh

```diff
@@ -207,6 +208,220 @@ test_client_association_dummy(){
+        'echo "EVENT RRM-BEACON-REP-RECEIVED 11:22:33:44:55:66 channel=1 dialog_token=0 measurement_rep_mode=0 op_class=0 duration=50 rcpi=-50 rsni=30 bssid=00:11:22:33:00:10" > /tmp/$USER/beerocks/wlan0/EVENT'
+    sleep 1
+
+    dbg "Confirming 11k request is done"
+    check docker exec -it repeater1 sh -c \
+        "grep -i -q 'Beacon 11k request 11:22:33:44:55:66 00:11:22:33:00:20 149' /tmp/$USER/beerocks/logs/beerocks_monitor_wlan0.log"
+
+    dbg "RPM Beacon Response"
+    check docker exec -it repeater1 sh -c \
+        'echo "EVENT RRM-BEACON-REP-RECEIVED 11:22:33:44:55:66 channel=149 dialog_token=0 measurement_rep_mode=0 op_class=0 duration=50 rcpi=-40 rsni=40 bssid=00:11:22:33:00:20" > /tmp/$USER/beerocks/wlan0/EVENT'
+    sleep 1
+
+    dbg "Confirming Client Association Control Request message was received (UNBLOCK)"
+    check docker exec -it repeater1 sh -c \
+        'grep -i -q "Got client allow request" /tmp/$USER/beerocks/logs/beerocks_agent_wlan2.log'
+
+    dbg "Confirming Client Association Control Request message was received (BLOCK)"
+    check docker exec -it repeater1 sh -c \
+        'grep -i -q "Got client disallow request" /tmp/$USER/beerocks/logs/beerocks_agent_wlan0.log'
+
+    dbg "Confirming Client Association Control Request message was received (BLOCK)"
+    check docker exec -it repeater2 sh -c \
+        'grep -i -q "Got client disallow request" /tmp/$USER/beerocks/logs/beerocks_agent_wlan0.log'
+
+    dbg "Confirming Client Association Control Request message was received (BLOCK)"
+    check docker exec -it repeater2 sh -c \
+        'grep -i -q "Got client disallow request" /tmp/$USER/beerocks/logs/beerocks_agent_wlan2.log'
+
+    dbg "Confirming Client Steering Request message was received - mandate"
+    check docker exec -it repeater1 sh -c \
+        'grep -i -q "Got steer request" /tmp/$USER/beerocks/logs/beerocks_agent_wlan0.log'
+
+    dbg "Confirming BTM Report message was received"
+    check docker exec -it gateway sh -c \
+        'grep -i -q "CLIENT_STEERING_BTM_REPORT_MESSAGE" /tmp/$USER/beerocks/logs/beerocks_controller.log'
+
+    dbg "Confirming ACK message was received"
+    check docker exec -it repeater1 sh -c \
+        'grep -i -q "ACK_MESSAGE" /tmp/$USER/beerocks/logs/beerocks_agent_wlan0.log'
+
+    dbg "Disconnect dummy STA from wlan0"
+    check docker exec -it repeater1 sh -c \
+        'echo "EVENT AP-STA-DISCONNECTED 11:22:33:44:55:66" > /tmp/$USER/beerocks/wlan0/EVENT'
+
+    #TODO// check for "disconnected after successful steering, proceeding to unblock" message 
+
+    dbg "Connect dummy STA to wlan2"
+    check docker exec -it repeater1 sh -c \
+        'echo "EVENT AP-STA-CONNECTED 11:22:33:44:55:66" > /tmp/$USER/beerocks/wlan2/EVENT'
+
+    dbg "Confirm steering success by client connected"
+    check docker exec -it gateway sh -c \
+        'grep -i -q "steering successful for sta 11:22:33:44:55:66" /tmp/$USER/beerocks/logs/beerocks_controller.log'
```

Comment:
If I'm not mistaken, these tests are already done by other test flows, so isn't it redundant to do it again?
I think that it is enough to check that optimal_path has decided to steer the client, without checking the whole steering flow.
Another option is to use the functions that already test some flows (like steering) and combine them in the current test (as a function call I mean).

Reply:
Will revert this test and commit new one. To big diff to review.